### PR TITLE
DVC-5679 fix all variables

### PR DESCRIPTION
**Pull request by ajwootto**
Fix SDK always returning empty object for allVariables call.

Remove `VariableCollection` class, which was basically just a reimplementation of a Dictionary after Jason's changes.

### CODE REVIEW

**Review by jsalaber — APPROVED** [review]

**Review by elliotCamblor — APPROVED** [review]

File: DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs [hunk 1]
```diff
@@ -122,8 +122,8 @@ public Variable<T> Variable<T>(User user, string key, T defaultValue)
             {
                 try
                 {
-                    ReadOnlyVariable<object> readOnlyVariable = config.Variables.Get(key);
-                    existingVariable = new Variable<T>(readOnlyVariable, defaultValue);
+                    ReadOnlyVariable<object> readOnlyVariable = config.Variables[key];
```
Comment (JamieSinn) [inline DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs:125]: With direct dict access; does this still handle safely with a missing kv pair for the `key`?

Comment (JamieSinn) [reply DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs:125]: Though I guess this is now the same because we didn't use `GetOrDefault`

Comment (ajwootto) [reply DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs:125]: it's inside a `ContainsKey` check

Comment (JamieSinn) [reply DevCycle.SDK.Server.Local/Api/DVCLocalClient.cs:125]: context ain't the best on github.

**Review by JamieSinn — APPROVED** [review]

**Pull request merged**